packages/core/causalorder/testframework.go

```diff
@@ -2,35 +2,118 @@ package causalorder
 
 import (
 	"fmt"
+	"sync"
+	"testing"
 
+	"github.com/cockroachdb/errors"
+	"github.com/iotaledger/hive.go/core/debug"
 	"github.com/iotaledger/hive.go/core/generics/options"
+	"github.com/stretchr/testify/require"
 
 	"github.com/iotaledger/goshimmer/packages/core/epoch"
+	"github.com/iotaledger/goshimmer/packages/core/eviction"
 )
 
+// region TestFramework ////////////////////////////////////////////////////////////////////////////////////////////////
+
+// TestFramework is the test framework for the CausalOrder.
 type TestFramework struct {
-	entitiesByAlias map[string]*MockOrderedEntity
+	test                 *testing.T
+	idCounter            int
+	idCounterMutex       sync.Mutex
+	entitiesByAlias      map[string]*MockedOrderedEntity
+	orderedEntities      map[string]bool
+	orderedEntitiesMutex sync.RWMutex
+	evictedEntities      map[string]bool
+	evictedEntitiesMutex sync.RWMutex
+
+	*CausalOrder[MockedEntityID, *MockedOrderedEntity]
 }
 
 // NewTestFramework is the constructor of the TestFramework.
-func NewTestFramework(opts ...options.Option[TestFramework]) (newFramework *TestFramework) {
+func NewTestFramework(test *testing.T, opts ...options.Option[TestFramework]) (newFramework *TestFramework) {
 	return options.Apply(&TestFramework{
-		entitiesByAlias: make(map[string]*MockOrderedEntity),
-	}, opts)
+		test:            test,
+		entitiesByAlias: make(map[string]*MockedOrderedEntity),
+		orderedEntities: make(map[string]bool),
+		evictedEntities: make(map[string]bool),
+	}, opts, func(t *TestFramework) {
+		t.CausalOrder = New[MockedEntityID, *MockedOrderedEntity](
+			eviction.NewManager[MockedEntityID](func(id MockedEntityID) (isRootBlock bool) {
+				return id.id == 0
+			}),
+			func(id MockedEntityID) (entity *MockedOrderedEntity, exists bool) {
+				return t.Get(id.alias)
+			}, func(entity *MockedOrderedEntity) (isOrdered bool) {
+				return entity.ordered
+			}, func(entity *MockedOrderedEntity) (err error) {
+				entity.ordered = true
+
+				if debug.GetEnabled() {
+					t.test.Logf("%s ordered", entity.id.alias)
+				}
+
+				t.orderedEntitiesMutex.Lock()
+				t.orderedEntities[entity.id.alias] = true
+				t.orderedEntitiesMutex.Unlock()
+
+				return nil
+			}, func(entity *MockedOrderedEntity, reason error) {
+				entity.invalid = true
```

Comment:
Can this be called from different goroutines? If yes, then we might need a setter/getter for the `MockedOrderedEntity` properties